cips/cip-tx-limits.md

```diff
@@ -0,0 +1,44 @@
+| cip | TBD(27) |
+| - | - |
+| title | Block limits for number of PFBs/non-PFBs and transaction sizes |
+| description | Set limits for number of PFBs and non-PFBs per block and transaction size |
+| author | Josh Stein ([@jcstein](https://github.com/jcstein)) |
+| discussions-to |  <https://forum.celestia.org/t/cip-limit-number-of-pfbs-and-non-pfbs-per-block-increase-transaction-size-limit/1843> |
+| status | Draft |
+| type | Standards Track |
+| category | Core |
+| created | 2024-10-16 |
+
+## Abstract
+
+This CIP proposes to set limits for the number of PFBs and non-PFBs per block and set the limit for transaction size. Concretely, the proposal is to set the limits to 600 PFBs and 200 non-PFB transactions per block and set the transaction size limit to 2MiB.
+
+## Specification
+
+1. The number of PFBs per block is limited to 600 by setting `PFBTransactionCap` to 600. `PFBTransactionCap` is the maximum number of PFB messages a block can contain.
+
+1. The number of non-PFB transactions per block is limited to 200 by setting `NonPFBTransactionCap` to 200. `NonPFBTransactionCap` is the maximum number of SDK messages, aside from PFBs, that a block can contain.
+
+1. The size of a transaction is limited to 2MB by setting `MaxTxBytes` to 2091752, which is 2MiB in bytes.
```

Comment:
```suggestion
1. The size of a transaction is limited to 2MB by setting `MaxTxBytes` to 2091752, which is 2MiB in bytes. From version v3 and above, in `CheckTx`, `PrepareProposal`, and `ProcessProposal`, each transaction's size is checked against the `appconsts.MaxTxSize` threshold. This ensures that transactions over the limit are rejected or excluded at all stages, from initial submission to execution.
```

Reply:
https://github.com/celestiaorg/CIPs/pull/220/commits/f401653d08accfadc659e4d3b7ebe7bdc76f0123 added here because suggestion wasn't mergable, as it was on outdated version of PR